.ci/ci-check.py

```diff
@@ -83,6 +83,7 @@ def install_package(pkg: typing.Union[typing.List[str], Path]) -> typing.Generat
             "--refresh",
             "--sync",
             "--noconfirm",
+            "mingw-w64-x86_64-python-pip",
```

Comment:
How about adding `--needed` so that `pip` is only installed when it's actually not installed?
```suggestion
            "--noconfirm",
            "--needed",
            "mingw-w64-x86_64-python-pip",
```

Reply:
Done